tests/server/sockfilt.c

```diff
@@ -799,7 +809,8 @@ static int select_ws(int nfds, fd_set *readfds, fd_set *writefds,
     wsasock = curlx_sitosk(fd);
 
     /* check if the current internal handle was triggered */
-    if(wait != WAIT_FAILED && (wait - WAIT_OBJECT_0) <= i &&
+    if(wait != WAIT_FAILED && wait >= WAIT_OBJECT_0 &&
+       (wait - WAIT_OBJECT_0) >= 0 && (wait - WAIT_OBJECT_0) <= i &&
```

Comment:
What I was getting at before is the [return value](https://docs.microsoft.com/en-us/windows/win32/api/synchapi/nf-synchapi-waitformultipleobjectsex#return-value) has to be within range. `wait >= WAIT_OBJECT_0` is basically is unsigned >= 0 which is always true, so is `(wait - WAIT_OBJECT_0) >= 0`, so those can be removed. What was unclear to me is `(wait - WAIT_OBJECT_0) <= i` always within range which on further inspection it is because i is always < nfd.

Also I notice nfd is not necessarily the count of handles that you are passing, it may be nfd + 1 but that's not accounted for in the loop. I guess since it's the exit event that's omitted it's probably correct.

Reply:
> What was unclear to me is `(wait - WAIT_OBJECT_0) <= i` always within range which on further inspection it is because i is always < nfd.

So the original condition was alright, like I explained and there are no additional checks needed?





> I guess since it's the exit event that's omitted it's probably correct.

Yes, since I do not care for the exit event in the loop.

Reply:
> So the original condition was alright, like I explained and there are no additional checks needed?

Yes. `(wait - WAIT_OBJECT_0) <= i` is sufficient by itself because WAIT_OBJECT_0 is 0. If WAIT_OBJECT_0 was 1234 or something then other checks would be needed.